tools/codegen/pkg/schemacheck/generator.go

```diff
@@ -92,73 +93,80 @@ func (g *generator) GenGroup(groupCtx generation.APIGroupContext) error {
 	comparatorOptions.DisabledComparators = g.disabledComparators
 
 	if err := comparatorOptions.Validate(); err != nil {
-		return fmt.Errorf("could not validate comparator options: %w", err)
+		return nil, fmt.Errorf("could not validate comparator options: %w", err)
 	}
 
 	comparatorConfig, err := comparatorOptions.Complete()
 	if err != nil {
-		return fmt.Errorf("could not complete comparator options: %w", err)
+		return nil, fmt.Errorf("could not complete comparator options: %w", err)
 	}
 
+	var results []generation.Result
+
 	for _, version := range groupCtx.Versions {
 		klog.V(1).Infof("Verifying API schema for for %s/%s", groupCtx.Name, version.Name)
```

Comment:
I know you didn't change this line, but I noticed a double "for" here. Not something I would block merging this on but figured I would call it out since you are making changes in this area.
```suggestion
		klog.V(1).Infof("Verifying API schema for %s/%s", groupCtx.Name, version.Name)
```